pkg/k8shandler/forwarding.go

```diff
@@ -43,6 +43,11 @@ func isForwardingEnabled(cluster *logging.ClusterLogging) bool {
 
 func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config string, err error) {
 
+	if clusterRequest.cluster == nil || clusterRequest.cluster.Spec.Collection == nil {
+		logger.Warnf("unable to restart fluentd because of cluster collection is not specified")
```

Comment:
I would just make this debug... and it may be more appropriate to say something like "Collection is unspecified. Skipping collector config generation."

Reply:
Please see if you can add a test for this as well

Reply:
we should add `clusterRequest.cluster == nil` check in all `ClusterLoggingRequest` methods + method specific nil checks

Reply:
@vimalk78 I think the actual code has to be refactored to provide a single entry point where the CLO creates `clusterRequest`, like it's done for `kibana` in the CLO.